src/ovp_pipeline/_truth_helpers.py

```
@@ -44,6 +45,43 @@
 ] = {}
+# Per-cache-key locks.  ``_EVOLUTION_LOCK_REGISTRY_LOCK`` guards
+# *just* the registry's own insert path — once a per-key lock
+# exists, every miss for the same key acquires that lock and
+# blocks ONLY the duplicate.  Misses for different keys
+# (different vault / pack scope / object filter) proceed in
+# parallel.  rev-bot 208 round-2 #9 caught that a single global
+# lock would block unrelated pack-scoped or object-scoped misses
+# behind the full-vault prewarm's ~4-min compute.
+_EVOLUTION_CANDIDATE_LOCK_REGISTRY: dict[
+    tuple[str, tuple[tuple[str, int, int], ...], str, tuple[str, ...]],
+    "threading.Lock",
```

Ruff (0.15.12)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)